protobuf/syft_proto/execution/v1/protocol.proto

```diff
@@ -4,16 +4,16 @@ package syft_proto.execution.v1;
 option java_package = "org.openmined.syftproto.execution.v1";
 
 import "syft_proto/types/syft/v1/id.proto";
+import "syft_proto/execution/v1/role.proto";
 
 message Protocol {
     syft_proto.types.syft.v1.Id id = 1;
-    repeated string tags = 2;
-    string description = 3;
-    bool workers_resolved = 4;
-    repeated PlanAssignment plan_assignments = 5;
-}
-
-message PlanAssignment {
-    syft_proto.types.syft.v1.Id plan_id = 1;
-    syft_proto.types.syft.v1.Id worker_id = 2;
-}
+    string name = 2;
+    map<string, syft_proto.execution.v1.Role> roles = 3;
+    repeated string input_repartition = 4;
+    repeated string output_repartition = 5;
+    bool include_state = 6;
+    bool is_built = 7;
```

Comment:
Can an unbuilt `Protocol` be serialized? I guess that train of thought applies to `Plans` too. 🤔

Reply:
Not at all, I thought the same thing while writing that! I can remove it for both Plan and Protocol.
(I think you made me shy about doing lots of refactoring/cleaning in a single PR 😄)